src/NzbDrone.Core/MediaFiles/ScriptImportDecider.cs

```diff
@@ -25,23 +28,75 @@ public class ImportScriptService : IImportScript
         private readonly IProcessProvider _processProvider;
         private readonly IConfigService _configService;
         private readonly ITagRepository _tagRepository;
+        private readonly IDiskProvider _diskProvider;
         private readonly Logger _logger;
 
         public ImportScriptService(IProcessProvider processProvider,
                                    IVideoFileInfoReader videoFileInfoReader,
                                    IConfigService configService,
                                    IConfigFileProvider configFileProvider,
                                    ITagRepository tagRepository,
+                                   IDiskProvider diskProvider,
                                    Logger logger)
         {
             _processProvider = processProvider;
             _videoFileInfoReader = videoFileInfoReader;
             _configService = configService;
             _configFileProvider = configFileProvider;
             _tagRepository = tagRepository;
+            _diskProvider = diskProvider;
             _logger = logger;
         }
 
+        private static readonly Regex MediaFileLine = new Regex(@"^\[SonarrMediaFile\]\s*(?<mediaFileName>.*)$", RegexOptions.Compiled);
+
+        private static readonly Regex ExtraFileLine = new Regex(@"^\[SonarrExtraFile\]\s*(?<extraFileName>.*)$", RegexOptions.Compiled);
+
+        private ScriptImportInfo ProcessStdout(List<ProcessOutputLine> processOutputLines, string defaultMediaFile)
+        {
+            var possibleExtraFiles = new List<string>();
+            string mediaFile = null;
+
+            foreach (var line in processOutputLines)
+            {
+                if (MediaFileLine.Match(line.Content) is var match && match.Success)
```

Comment:
With the suggestion to use one regex this will need to be reworked a bit, but should be pretty similar.

Reply:
I did indeed...... Fixed.